dev/com.ibm.ws.container.service.compat/src/com/ibm/ws/util/BoundedBuffer.java

```diff
@@ -975,6 +994,17 @@ protected synchronized boolean cancel(Object x) {
         return false;
     }
 
+    @Override
+    public void setAvailableProcessors(int availableProcessors) {
+        if (!spinsTakeProp) {
+            SPINS_TAKE_.set(availableProcessors - 1);
+        }
+
+        if (!spinsPutProp) {
```

Comment:
This if should be inside the other if.  If we aren't changing the SPINS_TAKE, there is nothing to do for the SPINS_PUT.